File: app/src/main/java/com/ninebx/ui/home/lists/adapter/RecyclerAdapter.java
package com.ninebx.ui.home.lists.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.ninebx.R;
import com.ninebx.ui.home.lists.model.AddedItem;

import java.util.ArrayList;

public class RecyclerAdapter extends RecyclerView.Adapter<RecyclerAdapter.RecyclerItemViewHolder> {
    private ArrayList<AddedItem> myList;
    int mLastPosition = 0;

    public RecyclerAdapter(ArrayList<AddedItem> myList) {
        this.myList = myList;
    }

    public RecyclerItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.row_list, parent, false);
        RecyclerItemViewHolder holder = new RecyclerItemViewHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(RecyclerItemViewHolder holder, final int position) {
        holder.etTitleTextView.setText(myList.get(position).getStrAddedItem());
        mLastPosition = position;
    }

    @Override
    public int getItemCount() {
        return (null != myList ? myList.size() : 0);
    }

    public void notifyData(ArrayList<AddedItem> myList) {
        this.myList = myList;
        notifyDataSetChanged();
    }

    public class RecyclerItemViewHolder extends RecyclerView.ViewHolder {
        private final EditText etTitleTextView;

        public RecyclerItemViewHolder(final View parent) {
            super(parent);
            etTitleTextView = (EditText) parent.findViewById(R.id.edtAddedList);
        }
    }
}

File: app/src/main/java/com/ninebx/ui/home/lists/adapter/RecyclerAdapter.kt
package com.ninebx.ui.home.lists.adapter

import android.annotation.SuppressLint
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.TextView

import com.ninebx.R
import com.ninebx.ui.home.lists.model.AddedItem

import java.util.ArrayList

class RecyclerAdapter(private var myList: ArrayList<AddedItem>?) : RecyclerView.Adapter<RecyclerAdapter.RecyclerItemViewHolder>() {
    internal var mLastPosition = 0

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerItemViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.row_list, parent, false)
        return RecyclerItemViewHolder(view)
    }


    override fun onBindViewHolder(holder: RecyclerItemViewHolder, @SuppressLint("RecyclerView") position: Int) {
        holder.etTitleTextView.text = myList!![position].strAddedItem
        mLastPosition = position
    }

    override fun getItemCount(): Int {
        return if (null != myList) myList!!.size else 0
    }

    fun notifyData(myList: ArrayList<AddedItem>) {
        this.myList = myList
        notifyDataSetChanged()
    }

    internal inner class RecyclerItemViewHolder(parent: View) : RecyclerView.ViewHolder(parent) {
        private val etTitleTextView: TextView
        private val layoutAddedList: RelativeLayout

        init {
            etTitleTextView = parent.findViewById<View>(R.id.txtListAdded) as TextView
            layoutAddedList = parent.findViewById<View>(R.id.layoutAddedList) as RelativeLayout
        }
    }

    fun add(location: Int, iName: String) {
        notifyItemInserted(location)
    }
}